### ISSUE TITLE
fitBounds not being updated when changing fitBoundsOptions

### ISSUE THREAD

**Issue by rofrol** [alex3165/react-mapbox-gl#355]
```jsx
import React, { Component } from 'react';
import ReactMap, { Marker } from 'react-mapbox-gl';

const accessToken = "pk.eyJ1IjoiYWxleDMxNjUiLCJhIjoiY2o0MHp2cGtiMGFrajMycG5nbzBuY2pjaiJ9.QDApU0XH2v35viSwQuln5w";
const style = "mapbox://styles/mapbox/streets-v9";

const BOUNDS = [
  [88.537838, 26.564062],
  [92.325374, 28.488102]
];

const CENTER = [90.2361995, 27.4128857];

const Map = ReactMap({
  accessToken
});

const mapStyle = {
  height: '100vh',
  width: '100vw'
};

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { fitBoundsOptions: { offset: [640 / 2, 0] }};
    this.setOffset = this.setOffset.bind(this);
  }
  setOffset(e) {
    let offset = [ parseInt(e.target.value, 10) / 2, 0];
    this.setState({ fitBoundsOptions:  { offset } });
  }
  render() {
    return (
      <div>
       Sidebar
       <input type="number" step="2" onChange={this.setOffset} />,
        offset[0]: {this.state.fitBoundsOptions.offset[0]}
       <Map
         style={style}
         containerStyle={mapStyle}
         fitBounds={BOUNDS}
         fitBoundsOptions={this.state.fitBoundsOptions}
       >
         <Marker
           coordinates={CENTER}
         >
           ---------------Center---------------
         </Marker>
       </Map>

      </div>
    );
  }
}

export default App;
```

https://www.webpackbin.com/bins/-KuV-ovZ4mCpiYXzbsnA

**Comment by rofrol** [comment]
I'm using the workaround to recreate the map:

```jsx
  componentWillReceiveProps (nextProps) {
    if (nextProps.sidebarWidth !== this.props.sidebarWidth) {
      // hack to rerender map https://github.com/alex3165/react-mapbox-gl/issues/355
      MapBox = ReactMapboxGl({ accessToken: mapBoxConfig.accessToken });
    }
  }
```

**Comment by alex3165** [comment]
Hey @rofrol this is an actual bug, map component don't update fitBounds if `fitBoundsOptions` changes. It is also a breaking change as people might not expect the fitBounds to change when they would break the reference of `fitBoundsOptions` between re-rendering.

**Comment by alex3165** [comment]
https://github.com/alex3165/react-mapbox-gl/pull/443 should fix it

**Comment by alex3165** [comment]
Released from tag `3.0.0`
